### PR TITLE
Fix `minikube tunnel` for hyperkit

### PR DESCRIPTION
**Pull request by balopat**
This change is a bit of a hack to make IP routing work over the virtio-net interface - it forces the virtio-net interface (bridge100) created by hyperkit/xhyve to reset via removing and readding the single member of the interface. Without this "reset" packets are just simply lost most of the time, despite a completely healthy IP route. 

This error was also the reason for a large percentage of our integration test failures. While reducing the flakiness of tunnel related tests I also increased the timeout requirement and adding retries to the nginx request in `TestTunnel`.

### CODE REVIEW

File: test/integration/tunnel_test.go [hunk 1]
```diff
@@ -88,10 +88,16 @@ func testTunnel(t *testing.T) {
...
-	httpClient.Timeout = 1 * time.Second
-	resp, err := httpClient.Get(fmt.Sprintf("http://%s", nginxIP))
+	httpClient.Timeout = 5 * time.Second
 
-	if err != nil {
+	var resp *http.Response
+
+	request := func() error {
+		resp, err = httpClient.Get(fmt.Sprintf("http://%s", nginxIP))
+		return err
```
Comment (tstromberg) [inline test/integration/tunnel_test.go]: AFAIK, For util.Retry to be functional, this error has to be be wraped up into in a util.RetriableError.

Comment (balopat) [reply test/integration/tunnel_test.go]: ah, true, I forgot that small thing :) thanks!

**Comment by balopat** [comment]
@minikube-bot OK to test

**Comment by balopat** [comment]
@minikube-bot OK to test

**Comment by balopat** [comment]
@minikube-bot OK to test

File: pkg/minikube/tunnel/tunnel.go [hunk 2]
```diff
@@ -136,17 +141,53 @@ func setupRoute(t *tunnel) {
...
+			glog.Errorf("failed to register tunnel: %s", err)
+			t.status.RouteError = err
+			return
+		} else if h.DriverName == "hyperkit" {
+			//the virtio-net interface acts up with ip tunnels :(
+			command := exec.Command("ifconfig", "bridge100")
+			fmt.Printf("About to run command: %s\n", command.Args)
+			response, _ := command.CombinedOutput()
+			iface := string(response)
+			pattern := regexp.MustCompile(`.*member: (en\d+) flags=3<LEARNING,DISCOVER>.*`)
```
Comment (tstromberg) [inline pkg/minikube/tunnel/tunnel.go]: If your going to use "Must*", make it a package var.

Also, including flag values makes this regexp very fragile and susceptible to OS update churn. In lieu of someone making SCNetworkConfiguration available in Go, I wonder if it's possible to get the ifconfig, ipconfig, or networksetup commands to more simply list member interfaces. 

If not, the regexp approach is probably OK, but I think `member: en\d+` might be enough.

File: pkg/minikube/tunnel/tunnel.go [hunk 3]
```diff
@@ -136,17 +141,53 @@ func setupRoute(t *tunnel) {
...
+			submatch := pattern.FindStringSubmatch(iface)
+			if len(submatch) != 2 {
+				//error
```
Comment (tstromberg) [inline pkg/minikube/tunnel/tunnel.go]: remove comment

File: pkg/minikube/tunnel/tunnel.go [hunk 4]
```diff
@@ -136,17 +141,53 @@ func setupRoute(t *tunnel) {
...
+				t.status.RouteError = fmt.Errorf("error, couldn't find member in bridge100 interface: \n%s", iface)
```
Comment (tstromberg) [inline pkg/minikube/tunnel/tunnel.go]: Remove "error," - as including "error," in the string this far down makes it quite likely that the final error message seen by the user will have "error" mentioned multiple times. 

Also, no need for a newline in this error message.

File: pkg/minikube/bootstrapper/kubeadm/templates.go [hunk 5]
```diff
@@ -110,6 +110,7 @@ WantedBy=multi-user.target
 `
 
 var kubeadmRestoreTemplate = template.Must(template.New("kubeadmRestoreTemplate").Parse(`
+sudo systemctl stop docker && sudo /sbin/minikube-automount && sudo systemctl start docker &&
```
[the remark this answers is not in the document]
Comment (balopat) [reply pkg/minikube/bootstrapper/kubeadm/templates.go]: ok, I'll try without it again - but I think this fixes an important flake when restart time the docker mount disappears. For some reason restarting docker causes the mount point /var/lib/docker to be dropped and when it comes back it recreates the /var/lib/docker directory on rootfs. It is not strictly related but it did fail my tests time to time.

File: pkg/minikube/tunnel/route_darwin.go [hunk 6]
```diff
@@ -48,10 +48,8 @@ func (router *osRouter) EnsureRouteIsAdded(route *Route) error {
 		return fmt.Errorf("error adding Route: %s, %d", message, len(strings.Split(message, "\n")))
```
[the remark this answers is not in the document]
Comment (balopat) [reply pkg/minikube/tunnel/route_darwin.go]: ah, that's a result of an old refactor that went wrong - I'll put in a fix for it here.

File: pkg/minikube/tunnel/route_darwin.go [hunk 7]
```diff
@@ -38,7 +38,7 @@ func (router *osRouter) EnsureRouteIsAdded(route *Route) error {
 	serviceCIDR := route.DestCIDR.String()
 	gatewayIP := route.Gateway.String()
 
-	glog.Infof("Adding Route for CIDR %s to gateway %s", serviceCIDR, gatewayIP)
+	glog.Infof("Adding routefor CIDR %s to gateway %s", serviceCIDR, gatewayIP)
```
Comment (tstromberg) [inline pkg/minikube/tunnel/route_darwin.go]: Add missing space, here and elsewhere in the route log statements.

Comment (balopat) [reply pkg/minikube/tunnel/route_darwin.go]: damn, automated refactoring and not paying attention to details! thanks for catching it :)

**Pull request merged**